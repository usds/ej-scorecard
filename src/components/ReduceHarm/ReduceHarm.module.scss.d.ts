declare namespace ReduceHarmNamespace {
  export interface IReduceHarmScss {
    summaryBox: string;
    h2: string;
  }
}

declare const ReduceHarmScssModule: ReduceHarmNamespace.IReduceHarmScss & {
  /** WARNING: Only available when "css-loader" is used without "style-loader" or "mini-css-extract-plugin" */
  locals: ReduceHarmNamespace.IReduceHarmScss;
};

export = ReduceHarmScssModule;
