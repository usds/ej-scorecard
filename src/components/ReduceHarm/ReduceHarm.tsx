import React from 'react';
import {
  Grid,
  SummaryBox,
  SummaryBoxContent,
  SummaryBoxHeading,
  Table,
} from '@trussworks/react-uswds';
import { ReduceHarmProps } from '@/types';

// import * as styles from './ReduceHarm.module.scss';

const ReduceHarm: React.FC<ReduceHarmProps> = ({ reduceHarm }) => {
  const table1Data = (
    <>
      <tbody>
        <tr>
          <td className="font-mono-sm text-tabular text-left">
            {reduceHarm.node.Row11}
          </td>
          <td className="font-mono-sm text-tabular text-left">
            {reduceHarm.node.Row12}
          </td>
        </tr>
        <tr>
          <td className="font-mono-sm text-tabular text-left">
            {reduceHarm.node.Row21}
          </td>
          <td className="font-mono-sm text-tabular text-left">
            {reduceHarm.node.Row22}
          </td>
        </tr>
        <tr>
          <td className="font-mono-sm text-tabular text-left vertical-align-top">
            {reduceHarm.node.Row31}
          </td>
          <td className="font-mono-sm text-tabular text-left">
            {reduceHarm.node.Row32}
          </td>
        </tr>
      </tbody>
    </>
  );

  return (
    <>
      <Grid row>
        <Grid desktop={{ col: 8 }} tablet={{ col: 10 }} col={12}>
          <h2 className={`scorecard-header2`}>Reducing burdens & harms</h2>
        </Grid>
      </Grid>
      <Grid row>
        <Grid desktop={{ col: 6 }} tablet={{ col: 10 }} col={12}>
          Short narrative about the why reducing hurdens & harms is important. I
          prem ins _dolor sit amet concertetur adinissing plitvam at magna vitan
          prat convall bandit. Phasellus id urna ultrices, vestibulum guam a.
          accumsan mauris
        </Grid>
      </Grid>
      <Grid row gap={6}>
        <Grid desktop={{ col: 4 }} tablet={{ col: 10 }} col={12}>
          <SummaryBox className={`margin-top-3`}>
            <SummaryBoxHeading headingLevel="h3">
              {`Reducing burdens & harms at ${reduceHarm.node.Agency_Name}`}
            </SummaryBoxHeading>
            <SummaryBoxContent>{reduceHarm.node.Row42}</SummaryBoxContent>
          </SummaryBox>
        </Grid>
        <Grid desktop={{ col: 8 }} tablet={{ col: 10 }} col={12}>
          <Table fullWidth bordered striped>
            {table1Data}
          </Table>
        </Grid>
      </Grid>
    </>
  );
};

export default ReduceHarm;
