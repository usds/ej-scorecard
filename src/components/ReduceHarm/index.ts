import ReduceHarm from './ReduceHarm';
export default ReduceHarm;
