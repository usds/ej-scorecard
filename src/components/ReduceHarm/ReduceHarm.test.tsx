import createComponentWithIntl from '@/test/testHelpers';

import ReduceHarm from './ReduceHarm';

const reduceHarm = {
  node: {
    id: `6575a3c4-0b44-51e4-867c-7eb615d76ced`,
    Agency_Name: `T2 Agency Dept Office`,
    Summary: `Lorem T2 ipsum dolor sit amet, consectetur adipiscing elit. Morbi luctus velit non justo tempus, sit amet bibendum tellus convallis. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia curae; Nam in gravida mi. Nam nisl ante, ultricies et massa id, euismod sodales arcu. Sed velit felis, tristique a nunc sit amet, rutrum iaculis lectus. Nullam auctor justo at odio sagittis rutrum. Nunc eget malesuada mauris. Donec tincidunt ex in mi ultrices sodales. Aliquam at lobortis eros, eu mollis tellus. Nullam sit amet elementum tellus. Suspendisse viverra ligula neque, id dapibus nisl elementum et. Integer mattis magna ac lorem fermentum, laoreet accumsan tellus tincidunt. Donec bibendum viverra velit vel eleifend.`,
    Row11: `Metric T2`,
    Row12: `#`,
    Row21: `Metric T2`,
    Row22: `#`,
    Row31: `Optional: Metric T2`,
    Row32: `Lorem T2 ipsum dolor sit amet, consectetur adipiscing elit. Morbi luctus velit non justo tempus, sit amet bibendum tellus convallis. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia curae; Nam in gravida mi. Nam nisl ante, ultricies et massa id, euismod sodales arcu. Sed velit felis, tristique a nunc sit amet, rutrum iaculis lectus. Nullam auctor justo at odio sagittis rutrum. Nunc eget malesuada mauris. Donec tincidunt ex in mi ultrices sodales. Aliquam at lobortis eros, eu mollis tellus. Nullam sit amet elementum tellus. Suspendisse viverra ligula neque, id dapibus nisl elementum et. Integer mattis magna ac lorem fermentum, laoreet accumsan tellus tincidunt. Donec bibendum viverra velit vel eleifend.`,
  },
};

describe(`rendering of ReduceHarm Component`, () => {
  it(`checks if component renders`, () => {
    const component = createComponentWithIntl(
      <ReduceHarm reduceHarm={reduceHarm} />,
    );

    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
