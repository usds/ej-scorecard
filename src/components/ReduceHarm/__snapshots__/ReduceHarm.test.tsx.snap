// Vitest Snapshot v1

exports[`rendering of ReduceHarm Component > checks if component renders 1`] = `
[
  <div
    className="grid-row"
    data-testid="grid"
  >
    <div
      className="grid-col-12 tablet:grid-col-10 desktop:grid-col-8"
      data-testid="grid"
    >
      <h2
        className="scorecard-header2"
      >
        Reducing burdens & harms
      </h2>
    </div>
  </div>,
  <div
    className="grid-row"
    data-testid="grid"
  >
    <div
      className="grid-col-12 tablet:grid-col-10 desktop:grid-col-6"
      data-testid="grid"
    >
      Short narrative about the why reducing hurdens & harms is important. I prem ins _dolor sit amet concertetur adinissing plitvam at magna vitan prat convall bandit. Phasellus id urna ultrices, vestibulum guam a. accumsan mauris
    </div>
  </div>,
  <div
    className="grid-row grid-gap-6"
    data-testid="grid"
  >
    <div
      className="grid-col-12 tablet:grid-col-10 desktop:grid-col-4"
      data-testid="grid"
    >
      <div
        className="usa-summary-box margin-top-3"
        data-testid="summary-box"
      >
        <div
          className="usa-summary-box__body"
        >
          <h3
            className="usa-summary-box__heading"
          >
            Reducing burdens & harms at T2 Agency Dept Office
          </h3>
          <div
            className="usa-summary-box__text"
          >
            Lorem T2 ipsum dolor sit amet, consectetur adipiscing elit. Morbi luctus velit non justo tempus, sit amet bibendum tellus convallis. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia curae; Nam in gravida mi. Nam nisl ante, ultricies et massa id, euismod sodales arcu. Sed velit felis, tristique a nunc sit amet, rutrum iaculis lectus. Nullam auctor justo at odio sagittis rutrum. Nunc eget malesuada mauris. Donec tincidunt ex in mi ultrices sodales. Aliquam at lobortis eros, eu mollis tellus. Nullam sit amet elementum tellus. Suspendisse viverra ligula neque, id dapibus nisl elementum et. Integer mattis magna ac lorem fermentum, laoreet accumsan tellus tincidunt. Donec bibendum viverra velit vel eleifend.
          </div>
        </div>
      </div>
    </div>
    <div
      className="grid-col-12 tablet:grid-col-10 desktop:grid-col-8"
      data-testid="grid"
    >
      <table
        className="usa-table src-components-Table-Table-module__fullwidth--fOyj6 usa-table--striped"
        data-testid="table"
      >
        <tbody>
          <tr>
            <td
              className="font-mono-sm text-tabular text-center"
            >
              Metric T2
            </td>
            <td
              className="font-mono-sm text-tabular text-center"
            >
              #
            </td>
          </tr>
          <tr>
            <td
              className="font-mono-sm text-tabular text-center"
            >
              Metric T2
            </td>
            <td
              className="font-mono-sm text-tabular text-center"
            >
              #
            </td>
          </tr>
          <tr>
            <td
              className="font-mono-sm text-tabular text-center"
            >
              Optional: Metric T2
            </td>
            <td
              className="font-mono-sm text-tabular text-center"
            >
              Lorem T2 ipsum dolor sit amet, consectetur adipiscing elit. Morbi luctus velit non justo tempus, sit amet bibendum tellus convallis. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia curae; Nam in gravida mi. Nam nisl ante, ultricies et massa id, euismod sodales arcu. Sed velit felis, tristique a nunc sit amet, rutrum iaculis lectus. Nullam auctor justo at odio sagittis rutrum. Nunc eget malesuada mauris. Donec tincidunt ex in mi ultrices sodales. Aliquam at lobortis eros, eu mollis tellus. Nullam sit amet elementum tellus. Suspendisse viverra ligula neque, id dapibus nisl elementum et. Integer mattis magna ac lorem fermentum, laoreet accumsan tellus tincidunt. Donec bibendum viverra velit vel eleifend.
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>,
]
`;
